### PR TITLE
Modernize Project

### PR DESCRIPTION
**Pull request by kylecesmat**
Tracking a new PR from a project branch. Originally created in #11.

Todo:

Fixed:
- Proper linting
- Removed unused code
- #9 Support for React 15
- Document API changes
- Remove unused props
- Add default styles
- #3 Support for manual closing

### CODE REVIEW

**Comment by jpollard-cs** [comment]
what can I do to help expedite this? happy to code review if needed! I can't shrinkwrap my related project without the updated peer dependencies (or I need to roll back all of my React 15 changes :open_mouth: )

File: package.json [hunk 1]
```diff
@@ -19,12 +19,13 @@
...
+    "classnames": "^2.2.0",
+    "react-addons-css-transition-group": "^15.0.2"
   },
   "peerDependencies": {
-    "redux": "^2.0.0 || ^3.0.0",
-    "react-redux": "^2.0.0 || ^3.1.0 || ^4.0.0",
-    "react": "^0.13.3 || ^0.14.0 || ^15.0.0-rc || ^15.0"
+    "redux": "^3.4.0",
+    "react-redux": "^4.4.0",
+    "react": "^15.0.0"
```
Comment (jpollard-cs) [inline package.json]: @kylecesmat what is the reason for dropping previous versions here?

File: src/components/Notifs.js [hunk 2]
```diff
@@ -1,54 +1,58 @@
...
-    const componentStyles = forceNotifsStyles || !theme ? styles : {};
-    return (
-      <div className={classnames('notif-container', className)} style={componentStyles}>
-        <TransitionGroup transitionName="notif">
-          {items}
-        </TransitionGroup>
-      </div>
-    );
-  }
+function Notifs(props) {
```
Comment (jpollard-cs) [inline src/components/Notifs.js]: :+1:

File: src/components/Notifs.js [hunk 3]
```diff
@@ -1,54 +1,58 @@
...
+      kind={getter(notif, 'kind')}
+      componentClassName={componentClassName}
+      CustomComponent={CustomComponent}
+      onActionClick={onActionClick}
+      action={action}
+    />
+  ));
+
+  return (
+    <div className={classnames(`${componentClassName}__container`, className)} >
```
Comment (jpollard-cs) [inline src/components/Notifs.js]: I know it's a small package, but given the uses of classnames here I'm not sure if it's really a necessary dependency?

**Comment by jpollard-cs** [comment]
any update on this?

**Comment by jpollard-cs** [comment]
@kylecesmat any update on this? I'm rolling with no shrinkwrap right now :sob:

**Comment by kylecesmat** [comment]
@jpollard-cs lets get this rolled out today! Sorry for the hiatus.

File: src/components/Notifs.js (hunk 3 above)
Comment (kylecesmat) [inline src/components/Notifs.js]: Would agree, lets get rid of it

**Comment by kylecesmat** [comment]
Ready to merge, did I miss anything?

**Comment by jpollard-cs** [comment]
:tada: Looks great now, thanks!!

**Comment by kylecesmat** [comment]
@jpollard-cs I merged this, we'll just need @asaf to tag & publish to npm

**Comment by jpollard-cs** [comment]
@kylecesmat thanks!

**Pull request merged**